src/js/trove/filelib.js

```diff
@@ -27,6 +30,17 @@
       this.name = name;
       this.fd = fs.openSync(name, (append ? "a" : "w"));
     }
+    var checkISD = RUNTIME.getField(SD, "internal").checkISD;
+    var isISD = RUNTIME.getField(SD, "internal").isISD;
+    function makeDirs(baseDir, tree) {
+      if (!fs.existsSync(baseDir)) fs.mkdirSync(baseDir);
+      if (isISD(tree)) {
+        RUNTIME.ffi.toArray(RUNTIME.getField(tree, "keys-list").app()).forEach(function(sub) {
+          var kid = RUNTIME.getColonField(tree, "get-value").full_meth(tree, sub);
```

Comment:
Maybe worth a comment that this won't stack capture because of what we know about `immutable-string-dict`'s `get-value` method being flat? Like if this was changed to a generic `.get-value` dictionary interface, this wouldn't be safe.

Reply:
Yeah, possibly